utils/zerovec/src/ule/plain.rs

```diff
@@ -201,6 +204,44 @@ impl AsULE for u8 {
 // EqULE is true because u8 is its own ULE.
 unsafe impl EqULE for u8 {}
 
+// Safety (based on the safety checklist on the ULE trait):
+//  1. NonZeroU8 does not include any uninitialized or padding bytes.
+//  2. NonZeroU8 is aligned to 1 byte.
+//  3. The impl of validate_byte_slice() returns an error if any byte is not valid (0x00).
+//  4. The impl of validate_byte_slice() returns an error if there are leftover bytes (never).
+//  5. The other ULE methods use the default impl.
+//  6. NonZeroU8 byte equality is semantic equality
+unsafe impl ULE for NonZeroU8 {
```

Comment:
Suggestion: make a `macro_rules!` to generate the ULE and NicheBytes impls for all `NonZero*` types

Reply:
We can't, because the other NonZero types need ULE wrappers.